docs/tutorials/setup-opentelemetry.md

```diff
@@ -46,23 +48,68 @@ const sdk = new NodeSDK({
 sdk.start();
 ```
 
-In the `index.ts`, import this file **at the beginning**:
+Your probably won't need all the instrumentation inside `getNodeAutoInstrumentations()` so make sure to
+check the [documentation](https://www.npmjs.com/package/@opentelemetry/auto-instrumentations-node) and tweak it properly.
 
-```typescript
-import './instrumentation'; // Setup the OpenTelemetry instrumentation
+It's important to setup the NodeSDK and the automatic instrumentation **before** importing any library.
 
-// other imports and backend init...
-```
+This is why we will use the nodejs [`--require`](https://nodejs.org/api/cli.html#-r---require-module)
+flag when we start up the application.
 
-It's important to setup the NodeSDK and the automatic instrumentation **before** importing any library.
+In your `Dockerfile` add the `--require` flag which points to the `instrumentation.ts` file
+
+```Dockerfile
+FROM node:18-bookworm-slim
+...
+# More functionality goes here
+...
+WORKDIR /app
+RUN chown node:node /app
+USER node
+
+ENV NODE_ENV production
+
+COPY --chown=node:node .yarn ./.yarn
+COPY --chown=node:node .yarnrc.yml ./
+
+# We need the instrumentation file inside the Docker image so we can use it with --require
+// highlight-add-next-line
+COPY --chown=node:node packages/backend/instrumentation.ts ./
```

Comment:
This line should probably go right before the CMD, so that the docker caching works better, right? Also backstage should probably add COPY --link to this file. Its part of the best practices for dockerfiles

https://docs.docker.com/reference/dockerfile/#copy---link

Reply:
For this I just followed the current state of the Dockerfile. Maybe @freben knows more about this. For now I changed the Dockerfile so it will be up to the user to fit the commands where he wants.